share/gpodder/extensions/tagging.py

```diff
@@ -256,10 +332,31 @@ def get_audio(self, info, episode):
                 info['title'],
                 info['subtitle'],
                 info['genre'],
-                info['pubDate'],
-                cover)
+                info['pubDate'])
         return audio
 
+    # extract coverart from episode, if exists
+    def get_embeddedart(self, info, episode):
+        audio = self.get_audio(info, episode)
+        return audio.extract_coverart()
+
+    # takes a raw bytes obj, returns a raw bytes obj
+    def normalize_image(self, bytesimg, filetype):
+        size = int(self.container.config.episode_coverart_size)
+        with Image.open(BytesIO(bytesimg)) as img:
+            if max(img.height, img.width) > size:
+                # Retain ratio, the larger of height, width will be set to size
+                ratio = min(size / img.height, size / img.width)
+                newsize = (int(ratio * img.width), int(ratio * img.height))
+                out = img.resize(newsize)
+            else:
+                out = img.copy()
+
```

Comment:
I got an exception with youtube-dl (eg with https://www.youtube.com/@JellyRollAnimations/videos), where there is no embedded cover art and no channel cover => the stock green feed icon was used. It is a png.


```suggestion

        if filetype == 'JPEG' and img.mode != 'RGB':
          out = img.convert('RGB')
```

Reply:
````
1740851040.983219 [gpodder.extensions] ERROR: Error in /home/elelay/devel/gPodder/share/gpodder/extensions/tagging.py in on_episode_downloaded: cannot write mode RGBA as JPEG
Traceback (most recent call last):
  File "/usr/lib/python3.13/site-packages/PIL/JpegImagePlugin.py", line 665, in _save
    rawmode = RAWMODE[im.mode]
              ~~~~~~~^^^^^^^^^
KeyError: 'RGBA'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/elelay/devel/gPodder/src/gpodder/extensions.py", line 76, in handler
    cb_res = callback(*args, **kwargs)
  File "/home/elelay/devel/gPodder/share/gpodder/extensions/tagging.py", line 304, in on_episode_downloaded
    embed_img = self.normalize_image(embed_img, config_filetype)
  File "/home/elelay/devel/gPodder/share/gpodder/extensions/tagging.py", line 357, in normalize_image
    out = img.convert('RGB')
  File "/usr/lib/python3.13/site-packages/PIL/Image.py", line 2596, in save
    save_handler(self, fp, filename)
    ~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.13/site-packages/PIL/JpegImagePlugin.py", line 668, in _save
    raise OSError(msg) from e
OSError: cannot write mode RGBA as JPEG
````

Reply:
tested the fix: it works

Reply:
I can't reproduce the issue, that channel appears to have a channel cover image. But the fix seems reasonable enough and still works for me.